##                      _             _ 
##  _ __ ___   ___   __| |    ___ ___| |  mod_ssl
## | '_ ` _ \ / _ \ / _` |   / __/ __| |  Apache Interface to OpenSSL
## | | | | | | (_) | (_| |   \__ \__ \ |  www.modssl.org
## |_| |_| |_|\___/ \__,_|___|___/___/_|  ftp.modssl.org
##                      |_____|         
## ____________________________________________________________________________
##
## Annotated patch file: addon.patch
## Copyright (c) 1998-2007 Ralf S. Engelschall, All Rights Reserved. 
## Created on: 08-Feb-2008
##
## This file assembles changes to existing Apache source files
## between the original Apache and the patched variant. It can be
## automatically applied to a vanilla Apache source tree with the
## 'patch' tool to upgrade those files.  Each patch snippet is
## annotated with a short description.
##

+---------------------------------------------------------------------------
| Add an entry for mod_define.
+---------------------------------------------------------------------------
Index: src/Configuration.tmpl
--- src/Configuration.tmpl	11 May 2004 18:28:09 -0000	1.1.1.8
+++ src/Configuration.tmpl	11 May 2004 18:32:15 -0000	1.24
@@ -258,6 +313,11 @@
 
 AddModule modules/standard/mod_env.o
 
+## mod_define expands variables on arbitrary directive lines.
+## It requires Extended API (EAPI).
+
+# AddModule modules/extra/mod_define.o
+
 ##
 ## Request logging modules
 ##

+---------------------------------------------------------------------------
| Add more beautiful optic to the status page table..
+---------------------------------------------------------------------------
Index: src/modules/standard/mod_status.c
--- src/modules/standard/mod_status.c	8 Feb 2008 11:15:41 -0000	1.1.1.19
+++ src/modules/standard/mod_status.c	8 Feb 2008 11:17:39 -0000	1.16
@@ -440,12 +440,33 @@
 	    if (no_table_report)
 		ap_rputs("<p><hr><h2>Server Details</h2>\n\n", r);
 	    else
+#ifndef NO_PRETTYPRINT
+		ap_rputs("<p>\n\n<table bgcolor=\"#ffffff\" border=\"0\">"
+			"<tr bgcolor=000000>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Srv</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>PID</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Acc</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>M</b></font></td>"
+#ifndef NO_TIMES
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>CPU</b></font></td>"
+#endif
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>SS</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Req</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Conn</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Child</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Slot</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Host</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>VHost</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Request</b></td>"
+			"</tr>\n", r);      
+#else /* NO_PRETTYPRINT */
 #ifdef NO_TIMES
 		/* Allow for OS/2 not having CPU stats */
 		ap_rputs("<p>\n\n<table border=0><tr><th>Srv<th>PID<th>Acc<th>M\n<th>SS<th>Req<th>Conn<th>Child<th>Slot<th>Client<th>VHost<th>Request</tr>\n\n", r);
 #else
 		ap_rputs("<p>\n\n<table border=0><tr><th>Srv<th>PID<th>Acc<th>M<th>CPU\n<th>SS<th>Req<th>Conn<th>Child<th>Slot<th>Client<th>VHost<th>Request</tr>\n\n", r);
 #endif
+#endif /* NO_PRETTYPRINT */
 	}
 
 	for (i = 0; i < HARD_SERVER_LIMIT; ++i) {
@@ -574,6 +595,11 @@
 				vhost->server_hostname) : "(unavailable)");
 		    }
 		    else {		/* !no_table_report */
+#ifndef NO_PRETTYPRINT
+			ap_rprintf(r,"<tr bgcolor=\"#ffffff\">");
+#else
+			ap_rprintf(r,"<tr>");
+#endif
 			if (score_record.status == SERVER_DEAD)
 #ifdef TPF
                             if (kill(ps_record.pid, 0) == 0) {
@@ -586,12 +612,12 @@
                             } else
 #endif /* TPF */
 			    ap_rprintf(r,
-				"<tr><td><b>%d-%d</b><td>-<td>%d/%lu/%lu",
+				"<td><b>%d-%d</b><td>-<td>%d/%lu/%lu",
 				i, (int) ps_record.generation,
 				(int) conn_lres, my_lres, lres);
 			else
 			    ap_rprintf(r,
-				"<tr><td><b>%d-%d</b><td>%d<td>%d/%lu/%lu",
+				"<td><b>%d-%d</b><td>%d<td>%d/%lu/%lu",
 				i, (int) ps_record.generation,
 				(int) ps_record.pid, (int) conn_lres,
 				my_lres, lres);
@@ -651,6 +677,16 @@
 			    ap_rprintf(r,
 			     "<td>?<td nowrap>?<td nowrap>..reading.. </tr>\n\n");
 			else
+#ifndef NO_PRETTYPRINT
+			    ap_rprintf(r,
+			     "<td nowrap><font face=\"Arial,Helvetica\" size=\"-1\">%s</font>"
+			     "<td nowrap><font face=\"Arial,Helvetica\" size=\"-1\">%s</font>"
+			     "<td nowrap><font face=\"Arial,Helvetica\" size=\"-1\">%s</font>"
+			     "</tr>\n\n",
+			     score_record.client,
+			     vhost ? vhost->server_hostname : "(unavailable)",
+			     ap_escape_html(r->pool, score_record.request));
+#else
 			    ap_rprintf(r,
 			     "<td>%s<td nowrap>%s<td nowrap>%s</tr>\n\n",
 			     ap_escape_html(r->pool, score_record.client),
@@ -658,6 +694,7 @@
 				vhost->server_hostname) : "(unavailable)",
 			     ap_escape_html(r->pool,
                                             ap_escape_logitem(r->pool, score_record.request)));
+#endif
 		    }		/* no_table_report */
 		}			/* !short_report */
 	    }			/* if (<active child>) */

+---------------------------------------------------------------------------
| Add a hyperlink to the mod_define.html document.
+---------------------------------------------------------------------------
Index: htdocs/manual/mod/index.html.en
--- htdocs/manual/mod/index.html.en	7 May 2006 18:10:42 -0000	1.1.1.4
+++ htdocs/manual/mod/index.html.en	7 May 2006 18:12:44 -0000	1.5
@@ -106,6 +106,10 @@
       <dd>Support for Netscape-like cookies. Replaced in Apache 1.2
       by mod_usertrack</dd>
 
+      <dt><a href="mod_define.html">mod_define</a></dt>
+
+      <dd>Variable Definition for Arbitrary Directives</dd>
+
       <dt><a href="mod_digest.html">mod_digest</a> Apache 1.1 and
       up</dt>
 

+---------------------------------------------------------------------------
| Add a hyperlink for the Define directives.
+---------------------------------------------------------------------------
Index: htdocs/manual/mod/directives.html.en
--- htdocs/manual/mod/directives.html.en	7 May 2006 18:10:41 -0000	1.1.1.8
+++ htdocs/manual/mod/directives.html.en	7 May 2006 18:12:44 -0000	1.9
@@ -233,6 +233,8 @@
 
       <li><a href="core.html#defaulttype">DefaultType</a></li>
 
+      <li><a href="mod_define.html#define">Define</a>
+
       <li><a href="mod_access.html#deny">Deny</a></li>
 
       <li><a href="core.html#directory">&lt;Directory&gt;</a></li>
